hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirResourceDaoR4SearchWithElasticSearchIT.java

```diff
@@ -1532,16 +1532,163 @@ private IIdType withObservationWithQuantity(double theValue, String theSystem, S
+					"?_tag=http://example.com|not-existing-tag,http://example.com|one-more-not-existing-tag" +
+					"&_tag=http://example.com|other-not-existing-tag,http://example.com|anotherTag");
+
+				assertEquals(0, myCaptureQueriesListener.getSelectQueriesForCurrentThread().size(), "we build the bundle with no sql");
+				assertThat(allIds, empty());
+			}
+
+			@Test
+			public void AndOrCombinedSearch_failing_and_with_single_or() {
+				myTestDataBuilder.createObservation(List.of(
+					myTestDataBuilder.withObservationCode("http://example.com/", "theCode"),
+					myTestDataBuilder.withTag("http://example.com", "aTag"),
+					myTestDataBuilder.withTag("http://example.com", "anotherTag"))).getIdPart();
+
+				myCaptureQueriesListener.clear();
+				List<String> allIds = myTestDaoSearch.searchForIds("/Observation" +
+					"?_tag=http://example.com|not-existing-tag" +
+					"&_tag=http://example.com|other-not-existing-tag,http://example.com|anotherTag");
+
+				assertEquals(0, myCaptureQueriesListener.getSelectQueriesForCurrentThread().size(), "we build the bundle with no sql");
+				assertThat(allIds, empty());
+			}
+
+		}
+
+		@Nested
+		public class TagSecurityTests {
+
+			@Test
+			public void simpleSearch() {
+				String id = myTestDataBuilder.createObservation(List.of(
+					myTestDataBuilder.withObservationCode("http://example.com/", "theCode"),
+					myTestDataBuilder.withSecurity("http://example.com", "security-label"))).getIdPart();
+
+				myCaptureQueriesListener.clear();
+				List<String> allIds = myTestDaoSearch.searchForIds("/Observation?_security=http://example.com|security-label");
+
+				assertEquals(0, myCaptureQueriesListener.getSelectQueriesForCurrentThread().size(), "we build the bundle with no sql");
+				assertThat(allIds, contains(id));
+			}
+
+			@Test
+			public void andOrCombinedSearch() {
+				String id = myTestDataBuilder.createObservation(List.of(
+					myTestDataBuilder.withObservationCode("http://example.com/", "theCode"),
+					myTestDataBuilder.withSecurity("http://example.com", "security-label"),
+					myTestDataBuilder.withSecurity("http://example.com", "other-security-label"))).getIdPart();
+
+				myCaptureQueriesListener.clear();
+				List<String> allIds = myTestDaoSearch.searchForIds("/Observation" +
+					"?_security=http://example.com|non-existing-security-label,http://example.com|security-label" +
+					"&_security=http://example.com|other-non-existing-security-label,http://example.com|other-security-label");
+
+				assertEquals(0, myCaptureQueriesListener.getSelectQueriesForCurrentThread().size(), "we build the bundle with no sql");
+				assertThat(allIds, contains(id));
+			}
+		}
+
+
+		@Nested
+		public class TagProfileTests {
+
+			@Test
+			public void simpleSearch() {
+				String id = myTestDataBuilder.createObservation(List.of(
+					myTestDataBuilder.withObservationCode("http://example.com/", "theCode"),
+					myTestDataBuilder.withProfile("http://example.com|theProfile"))).getIdPart();
```

Comment:
This is a funny look for a profile.  Profiles are usually plain URIs.
```suggestion
					myTestDataBuilder.withProfile("http://example.com/someProfile/v1"))).getIdPart();
```

Reply:
Changed